packages/bee-q/src/components/option/bq-option.tsx

```diff
@@ -0,0 +1,178 @@
+  @State() isOptionInGroup: boolean = false;
+
+  // Public Property API
+  // ========================
+
+  /** If true, the `bq-option` is disabled. */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** A string representing the value of the option. */
+  @Prop({ reflect: true }) value?: string;
+
+  /** If true, the option is selected and active. */
+  @Prop({ reflect: true }) selected: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when item loses focus */
+  @Event() bqBlur: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when item is focused */
+  @Event() bqFocus: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when item is clicked */
+  @Event() bqClick: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key press */
+  @Event() bqOnEnter: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key up */
+  @Event() bqOnEnterKeyUp: EventEmitter<HTMLBqOptionElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent) {
+    if (event.key === 'Enter') {
+      this.bqOnEnter.emit(this.el);
+      this.selected = true;
+    }
+  }
+
+  @Listen('keyup')
+  onKeyUp(event: KeyboardEvent) {
+    if (event.key === 'Enter') this.bqOnEnterKeyUp.emit(this.el);
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  @Method()
+  async setPaddingToOption() {
+    this.isOptionInGroup = true;
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private onBlur = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqBlur.emit(this.el);
+  };
+
+  private onFocus = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private onClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqClick.emit(this.el);
+  };
+
+  private onSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem, 'prefix');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div
```

Comment:
Wouldn't be better if you used `button` instead of `div` ?

Reply:
I didn't think of it as a `button` specifically, but as a list of option items, having the `role="option"` attribute.
But if you think it is more semantic to have a button, then I'll change it.